libsolidity/codegen/ir/Common.cpp

```diff
@@ -122,3 +127,20 @@ string IRNames::zeroValue(Type const& _type, string const& _variableName)
 {
 	return "zero_" + _type.identifier() + _variableName;
 }
+
+string sourceLocationComment(langutil::SourceLocation const& _location, IRGenerationContext const& _context)
+{
```

Comment:
```suggestion
{
    solAssert(_context.sourceIndices().count(_location.source->name()), "");
```
Otherwise, it maybe that the `.at(...)` will throw and we don't know where that one was coming from.

Reply:
I didn't feel the need to assert that because we literally iterate the sourceIndices map so it should be impossible for a lookup to fail here...

Reply:
I'm in general not that keen to avoid using `.at` - ok, it does not add a line number, but it creates an exception and you can plug it into a debugger to see the line.